framework/src/engine/consensus/synchronizer/block_synchronization_mechanism.ts

```diff
@@ -381,14 +375,12 @@ export class BlockSynchronizationMechanism extends BaseSynchronizer {
 					blockHeaders.map(block => block.id),
 				);
 			} catch (e) {
-				numberOfRequests += 1;
+				currentRound -= blocksPerRequestLimit;
+				currentHeight = currentRound * validators.length;
```

Comment:
why is it updating the current height in the error case?
Also what happen if `_getHighestCommonBlockFromNetwork` fails with network reason?